### PR TITLE
improve self-signed host list

### PR DESCRIPTION
**Pull request by xujinzheng**

### CODE REVIEW

File: transport/http_transport.go [hunk 1]
```diff
@@ -409,7 +409,16 @@ func (h *httpTransport) Listen(addr string, opts ...ListenOption) (Listener, err
 
 		fn := func(addr string) (net.Listener, error) {
 			if config == nil {
-				cert, err := mls.Certificate(addr)
+				hosts := []string{addr}
+				if h, _, e := net.SplitHostPort(addr); e == nil {
+					if h == "" {
```
Comment (asim) [inline transport/http_transport.go]: Change this to `len(h) == 0`

File: transport/http_transport.go [hunk 2]
```diff
@@ -447,3 +456,41 @@ func newHTTPTransport(opts ...Option) *httpTransport {
 	}
 	return &httpTransport{opts: options}
 }
+
+func getIPAddrList() []string {
```
Comment (asim) [inline transport/http_transport.go]: Rename to `getIPAddrs`

File: transport/http_transport.go [hunk 3]
```diff
@@ -447,3 +456,41 @@ func newHTTPTransport(opts ...Option) *httpTransport {
...
+	ifaces, err := net.Interfaces()
+
```
Comment (asim) [inline transport/http_transport.go]: Strip whitespace here

File: transport/http_transport.go [hunk 4]
```diff
@@ -447,3 +456,41 @@ func newHTTPTransport(opts ...Option) *httpTransport {
...
+	if err != nil {
+		return nil
+	}
+
+	var ipAddrlist []string
+
+	for _, i := range ifaces {
+		if addrs, err := i.Addrs(); err != nil {
+			continue
+		} else {
```
Comment (asim) [inline transport/http_transport.go]: Remove `else`. Do 

```
addrs, err := i.Addrs()
if err != nil {
  continue
}

for _, addr := range addrs {
...
```

**Comment by xujinzheng** [comment]
Done, and I will try `go vet` command at next PR.

**Pull request merged**